ibis/expr/operations/relations.py

```diff
@@ -531,6 +528,10 @@ def __init__(self, table, metrics, by, having, predicates, sort_keys):
             sort_keys=sort_keys,
         )
 
+    @attribute
+    def _projection(self):
+        return Projection(self.table, self.metrics + self.by)
```

Comment:
Maybe this should be `by + metrics` since that's the order we return them in to the user?

```suggestion
        return Projection(self.table, self.by + self.metrics)
```

Reply:
Yes, though this will matter when we split the Selection and Aggregation objects.